addons/ai_hearing/functions/fnc_initializeAIHearing.sqf

```diff
@@ -3,21 +3,25 @@
 /*
     Author(s):
         By Dimitri Yuri edited by 2600K
-        Dedmen
+        Dedmen, Dorbedo
 
     Description:
         Add Eventhandler that notifies nearby AI's when player is Speaking
 */
 
-if (!hasInterface) exitWith {}; //Only on clients
+if ((!hasInterface)||{!TFAR_AICanHearPlayer}||{missionNamespace getVariable [QGVAR(isInitialized),false]}) exitWith {}; //Only on clients
+
+GVAR(isInitialized) = true;
+
 
 ["TFAR_AI_Detection", "OnSpeak", {
+    If !(TFAR_AICanHearPlayer) exitWith {};
```

Comment:
I would rather just have the setting call removeEventhandler when it get's disabled.

Reply:
But this event only fires on the player so.. Probably doesn't really matter if we have an additional check here.
But Setting should still removeEH if GVAR(isInitialized)

Reply:
I would not recomend this to be enabled and disabled during the mission anyways.
I'll change it.